### PR TITLE
Expose InteropObservable and InteropSubscribable

### PR DESCRIPTION
**Pull request by davidkpiano**
closes https://github.com/statelyai/xstate/issues/4711

### CODE REVIEW

**Comment by davidkpiano** [comment]
@Andarist Can you quickly check if any other types I marked as `@internal` should really be `@hidden`? Or is it most of them?

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -1941,7 +1935,7 @@ export interface ActorRef<
    */
   id: string;
   sessionId: string;
-  /** @internal */
+  /** @hidden */
```
Comment (Andarist) [inline packages/core/src/types.ts:1938]:
```suggestion
  /** @internal */
```

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -1952,7 +1946,7 @@ export interface ActorRef<
   // TODO: figure out how to hide this externally as `sendTo(ctx => ctx.actorRef._parent._parent._parent._parent)` shouldn't be allowed
   _parent?: ActorRef<any, any>;
   system: AnyActorSystem;
-  /** @internal */
+  /** @hidden */
```
Comment (Andarist) [inline packages/core/src/types.ts:1949]:
```suggestion
  /** @internal */
```

File: packages/core/src/types.ts [hunk 3]
```diff
@@ -1688,7 +1688,7 @@ export interface StateConfig<
   context: TContext;
   historyValue?: HistoryValue<TContext, TEvent>;
   /**
-   * @internal
+   * @hidden
```
Comment (Andarist) [inline packages/core/src/types.ts:1691]:
```suggestion
   * @internal
```

File: .changeset/slow-swans-punch.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'xstate': patch
+---
+
+Ensure that `InteropObservable` and `InteropSubscribable` (and others) are present in the type definition file.
```
Comment (Andarist) [inline .changeset/slow-swans-punch.md:5]:
```suggestion
Ensure that `InteropObservable` and `InteropSubscribable` are present in the type definition file.
```

**Comment by Andarist** [comment]
I only identified `MachineImplementationsSimplified` as one extra type that was erroneously marked as `@internal`

**Comment by davidkpiano** [comment]
Changes applied; good to go?

**Pull request merged**